lib/elixir/lib/kernel.ex

```diff
@@ -1473,8 +1473,8 @@ defmodule Kernel do
 
   ## Implemented in Elixir
 
-  defp optimize_boolean({:case, meta, args}) do
-    {:case, [{:optimize_boolean, true} | meta], args}
+  defp optimize_boolean({:case, meta, args}, generated \\ true) do
+    {:case, [{:optimize_boolean, generated: generated} | meta], args}
```

Comment:
Sorry but I do not understand this change. Why are we defining such tuple with tuples inside?

Reply:
To clarify, I think we should be able to not change this, as conds compile as a series of nested case clauses, and our optimization is enough to handle case clauses, no? :)

Reply:
Maybe it's not clear enough, the problem with this clause is that in the optimization during the expansion the clauses are marked with `generated: true`, for `cond` this is a problem because of (this)[https://github.com/elixir-lang/elixir/blob/master/lib/elixir/test/elixir/kernel/warning_test.exs#L609] warning.

I can switch this to an atom (but I couldn't find a good one) or bring the generated to Kernel, leaving only the optimization to be done in Erlang.

Reply:
> I can switch this to an atom (but I couldn't find a good one) or bring the generated to Kernel, leaving only the optimization to be done in Erlang.

I like this. Two questions then:

1. What happens if we don't mark them as generated?
2. If something bad happens, then let's bring it to kernel.

Reply:
It creates a bunch of warnings:
```
warning: this clause cannot match because a previous clause always matches
```

I'll move it to kernel